tuf/client/handlers.py

```diff
@@ -92,64 +99,26 @@ class RemoteMetadataUpdater(MetadataUpdater):
   """
 
 
-  def get_mirrors(self, remote_filename):
-    """
-    <Purpose>
-      Finds mirrors from which the specified file can be downloaded.
-
-
-    <Arguments>
-      remote_filename:
-        The relative file path (on the remote repository) of a metadata role.
-
-
-    <Exceptions>
-      None.
-
-    Side Effects>
-      None.
-
-    <Returns>
-      A list of mirrors from which the specified file can be downloaded.
-    """
-    return tuf.mirrors.get_list_of_mirrors('meta', remote_filename,
-      self.mirrors)
-
+  def get_mirrors(self, **kwargs):
+    file_type = kwargs['file_type']
+    file_path = kwargs['file_path']
+    return tuf.mirrors.get_list_of_mirrors(file_type, file_path,
+        self.mirrors)
 
-  def get_metadata_file(self, file_mirror, _filename, _upperbound_filelength):
-    """
-    <Purpose>
-      Downloads the metadata file from the provided mirror. Calls 'unsafe_download', which,
-      given the 'url' and 'required_length' of the desired file downloads the file and
-      returns its contents.
-
-
-    <Arguments>
-      file_mirror:
-        Mirror from which the file should be downloaded.
-
-      _filename:
-        The relative file path (on the remote repository) of a metadata role.
-
-      _upperbound_filelength:
-        An integer value representing the upper limit of the length of the file.
-
-    <Exceptions>
-      tuf.ssl_commons.exceptions.DownloadLengthMismatchError, if there was a
-      mismatch of observed vs expected lengths while downloading the file.
-
-      securesystemslib.exceptions.FormatError, if any of the arguments are
-      improperly formatted.
-
-      Any other unforeseen runtime exception.
-
-    Side Effects>
-      A 'securesystemslib.util.TempFile' object is created on disk to store the
-      contents of 'url'.
-
-    <Returns>
-      A 'securesystemslib.util.TempFile' file-like object that points to the
-      contents of 'url'.
-    """
+  def get_metadata_file(self, file_mirror, **kwargs):
+    upperbound_filelength = kwargs['upperbound_filelength']
     return tuf.download.unsafe_download(file_mirror,
-        _upperbound_filelength)
+        upperbound_filelength)
+
+  def get_target_file(self, file_mirror, **kwargs):
+    download_safely = kwargs['download_safely']
```

Comment:
`kwargs.pop('download_safely', True)` if not passed? Line 115. should also use `pop`

Reply:
Resolved all